### DISCUSSION TITLE
How to check which mode is using in vue-router4?

### DISCUSSION THREAD

**Discussion by tellok** [vuejs/router#1809] [Help and Questions]
in previous versions, we can check `router.mode === 'history'`, sort of, while in vue-router4, seems there is no alternative?

**Comment by posva** [comment]
What's the use case? In v4, history can be custom made.

**Reply by tellok** [reply]
There is some platform limitations, we have to do some extra work in history mode, the problem is that we can't checkout the mode in rourter@4 now.
